### PR TITLE
Rework the message structure

### PR DESCRIPTION
**Pull request by robx**
This is meant as a base for a more serious EDN approach, but should clear things up a bit as is. I've tested that things still seem to work, might have missed certain message types though.

Also I'm not sure of the `WSMsg` organization, would you like me to collect those somewhere (where)?

### CODE REVIEW

File: elm/Mpd.elm [hunk 1]
```diff
@@ -248,3 +252,132 @@ dbentryDecoder =
...
+    | CmdTrack String String
+    | CmdFindAdd { artist : String, album : String, track : String }
+    | CmdLoadDir String String
+    | CmdSetVolume Float
+
+
+encodeCmd : WSCmd -> String
+encodeCmd cmd =
+    let
+        tag =
```
Comment (alicebob) [inline elm/Mpd.elm]: `let (tag, args) = case cmd of ...` ?

File: siren/websocket.go [hunk 2]
```diff
@@ -17,25 +17,9 @@ var upgrader = websocket.Upgrader{
...
-	What    string  `json:"what"`
-	Artist  string  `json:"artist"`
-	Album   string  `json:"album"`
-	Track   string  `json:"track"`
-	File    string  `json:"file"`
-	Seconds float64 `json:"seconds"`
-	Song    string  `json:"song"`
-	Volume  float64 `json:"volume"`
+type WSMsg interface {
+	isMsg()
```
Comment (alicebob) [inline siren/websocket.go]: maybe drop that? Seems not to really add value here.

Comment (robx) [reply siren/websocket.go]: just the `isMsg`? because we only use the `Type() string`?

Comment (alicebob) [reply siren/websocket.go]: yes. The isMsg is just Go type magic.

**Pull request merged**